frontend/components/puzzle7.jsx

```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+import styled from 'styled-components';
+
+import { smSpacing } from '../constants';
+import Switch from './switch';
+
+const SwitchContainer = styled.div`
+  display: flex;
+  flex-direction: row;
+  justify-content: space-around;
+  margin: ${smSpacing}px 0;
+`;
+
+const SubmitButton = styled.button`
+`;
+
+const INITIAL_SWITCH_STATE = [false, false, false, false, false, false, false, false, false, false];
+
+class Puzzle7 extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      switchState: INITIAL_SWITCH_STATE,
+    };
+    this.onToggleSwitch = this.onToggleSwitch.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  onToggleSwitch(index) {
+    return () => {
+      this.setState((state) => {
+        const newSwitchState = state.switchState.slice();
+        newSwitchState[index] = !newSwitchState[index];
+        return { switchState: newSwitchState };
```

Comment:
This is to check if the switchstate changed and if it did then update the component to new switchstate?

Reply:
This is triggered when someone presses a switch, and updates the state of the switches. Pretty similar to the button presses in puzzle 3 and 4